cuML/CMakeLists.txt

```diff
@@ -41,15 +41,15 @@ endif(ZLIB_FOUND)
 ###################################################################################################
 # - Submodules ------------------------------------------------------------------------------------
 
-set(GTEST_DIR ${PROJECT_SOURCE_DIR}/external/ml-prims/external/googletest CACHE STRING
```

Comment:
My bad. I should have mentioned this before.

I deliberately made this change. My final aim with this integration (and some subsequent refactorings) are to:
1. Have a single place to host (and also to refer) all of submodules required.
2. Remove duplicate submodules (which will also reduce clone time and repo size)
3. Make the top-level cmake file also build ml-prims tests. This will void the need for quite a bit of duplicated logic between cuML/CMakeLists.txt and ml-prims/CMakeLists.txt
4. Finally, this should also be helpful, for example, with the issue #82 .

Please bear with this temporary modification, until I fix those, right after this PR gets merged. What say?